lib/active_model/serializer/attributes.rb

```diff
@@ -55,6 +55,9 @@ def attributes(*attrs)
         #     end
         def attribute(attr, options = {}, &block)
           key = options.fetch(:key, attr)
+          # replace ? at the end of an attribute, as JSON API disallows the use
+          # of ? at the end of keys.
+          key = key.to_s.sub(/\?\z/, '').to_sym
```

Comment:
better if this were in a method someone could monkey patch if they absolutely hate it

better if it only applied in json api

best if it were part of the 'transform keys' logic for json api (which doesn't really exist right now)